src/idiomatic/leveraging-the-type-system/borrow-checker-invariants/generalizing-ownership.md

```diff
@@ -0,0 +1,46 @@
+---
+minutes: 0
+---
+
+The logic of the borrow checker, while modelled off "memory ownership", can be abstracted away from that use case to model other problems where we want to prevent API misuse.
+
+```rust,editable
+pub struct InternalData;
+pub struct Value(InternalData);
+
+fn shared_use(value: &Value) -> &InternalData {
+    &value.0
+}
+
+fn exclusive_use(value: &mut Value) -> &mut InternalData {
+    &mut value.0
+}
+
+fn deny_future_use(value: Value) {}
+
+let mut value = Value(InternalData);
+let deny_mut = shared_use(&value);
+let try_to_deny_immutable = exclusive_use(&mut value); // ❌🔨
+let more_mut_denial = &deny_mut;
+deny_future_use(value);
+let even_more_mut_denial = shared_use(&value); // ❌🔨
```

Comment:
In the speaker notes, we should elaborate as to what the example is trying to show and it relates to the concepts we're discussing.
This is a bit too implicit for me on this slide.